src/planner/binder/query_node/bind_select_node.cpp

```diff
@@ -55,13 +62,31 @@ unique_ptr<Expression> Binder::BindOrderExpression(OrderBinder &order_binder, un
 unique_ptr<BoundResultModifier> Binder::BindLimit(LimitModifier &limit_mod) {
 	auto result = make_unique<BoundLimitModifier>();
 	if (limit_mod.limit) {
-		result->limit = BindConstant(*this, context, "LIMIT clause", limit_mod.limit);
-		result->offset = 0;
+		auto limit_copy = limit_mod.limit->Copy();
+		if (!BindConstant(*this, context, "LIMIT clause", limit_copy,result->limit_val)){
+			Binder new_binder(context);
+			ExpressionBinder expr_binder(new_binder, context);
```

Comment:
Since you expect an unsigned number here, set the `target_type` of the ExpressionBinder to UBIGINT (or similar).